src/init.cpp

```diff
@@ -75,6 +75,9 @@ static const bool DEFAULT_STOPAFTERBLOCKIMPORT = false;
 std::unique_ptr<CConnman> g_connman;
 std::unique_ptr<PeerLogicValidation> peerLogic;
 
+/** Set at startup if the data directory was succesfully locked */
+bool g_locked_data_directory = false;
```

Comment:
static ok, atomic no, there's no concurrent modification as the flag is set from one place in the initialization sequence.